ELFSage/Util/List.lean

```diff
@@ -0,0 +1,36 @@
+namespace List
+
+-- from Std
+
+/--
+Pads `l : List α` with repeated occurrences of `a : α` until it is of length `n`.
+If `l` is initially larger than `n`, just return `l`.
+-/
+def leftpad (n : Nat) (a : α) (l : List α) : List α := replicate (n - length l) a ++ l
+
+-- from Mathlib
+
+section sort
+
+variable {α : Type uu} (r : α → α → Prop) [DecidableRel r]
+local infixl:50 " ≼ " => r
+
+section InsertionSort
```

Comment:
There's also a builtin quicksort:
https://github.com/leanprover/lean4/blob/baa4b68a71926a722b77d4ec2cba34bfc76cc5aa/src/Init/Data/Array/QSort.lean#L32-L44,
but I'm not quite sure of the tradeoffs (there are some theorems for insertion
sort, which is nice…)

Reply:
Ah nice. It looks like the built-in sorts are all on Array, not List. Copying less from Mathlib feels good.